### PR TITLE
feat: new overloads for worker server creation

### PR DESCRIPTION
**Pull request by aneojgurhem**

### CODE REVIEW

**Review by lemaitre-aneo — COMMENTED** [review]
What is the purpose of this?

File: packages/csharp/ArmoniK.Api.Worker/Utils/WorkerServer.cs [hunk 1]
```diff
@@ -48,6 +48,18 @@ namespace ArmoniK.Api.Worker.Utils;
...
+  /// <summary>
+  ///   Create a web application for the given ArmoniK Worker gRPC Service
+  /// </summary>
+  /// <typeparam name="T">gRPC Service to add to the web application</typeparam>
+  /// <returns>
+  ///   The web application initialized
+  /// </returns>
+  public static WebApplication Create<T>()
+    where T : gRPC.V1.Worker.Worker.WorkerBase
+    => Create<T>(null);
```
Comment (lemaitre-aneo) [inline packages/csharp/ArmoniK.Api.Worker/Utils/WorkerServer.cs:60]: This should be handled by the following function:

```c#
  public static WebApplication Create<T>(IConfiguration?             configuration       = null,
                                         Action<IServiceCollection>? serviceConfigurator = null)
```

Comment (lemaitre-aneo) [reply packages/csharp/ArmoniK.Api.Worker/Utils/WorkerServer.cs:60]: So you should most likely remove the optional first parameter to the other method.

Comment (aneojgurhem) [reply packages/csharp/ArmoniK.Api.Worker/Utils/WorkerServer.cs:60]: but it breaks backward compatibility

Comment (lemaitre-aneo) [reply packages/csharp/ArmoniK.Api.Worker/Utils/WorkerServer.cs:60]: It shouldn't. Default parameters are handled by the caller, not the callee. Have you tried?

Comment (aneojgurhem) [reply packages/csharp/ArmoniK.Api.Worker/Utils/WorkerServer.cs:60]: Yes. That's why I have the issue.  There is an unit test for that. You can try to remove the empty parameter overload and compile

**Review by lemaitre-aneo — APPROVED** [review]

**Pull request merged**